src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({ id, sender, body, timeStamp, onLikeChange }) => {
+  const [liked, setLiked] = useState(false);
```

Comment:
👀 We don't need (and should avoid introducing) an additional local piece of state to store whether the message this ChatEntry is displaying is liked. One of the values that should get passed in the `props` is the `liked` value, and that value (coming all the way from the state stored in App) should get updated when the like button is clicked. We can then calculate which heart emoji to display as follows:
```js
  const heart = liked ? '❤️' : '🤍';  // assumes liked is destructured as other props
```
We could calculate this local variable here, and use it in the like button below
```js
        <button className="like" onClick={handleLike}>{heart}</button>
```

Or perform the calculation in the layout (similar to the current approach, but use the `props` value instead)
```js
        <button className="like" onClick={handleLike}>{liked ? '❤️' : '🤍'}</button>
```